cmd/app/start/start.go

```diff
@@ -303,12 +303,12 @@ func (s *Start) CreateStartCommand() *cobra.Command {
 }
 
 func (s *Start) runE(cmd *cobra.Command, _ []string) error {
-	totalVulns, err := s.startAnalysis(cmd)
+	_, hasError, err := s.startAnalysis(cmd)
 	if err != nil {
 		return err
 	}
 
-	if totalVulns > 0 && s.configs.ReturnErrorIfFoundVulnerability {
+	if hasError && s.configs.ReturnErrorIfFoundVulnerability {
```

Comment:
You can use `if err != nil && s.configs.ReturnErrorIfFoundVulnerability` instead.

Reply:
Also, this code will not be executed, Since `hasError` is true when `err != nil` the code above ` if err != nil { return err }` will return the function and this if statement will never execute.  So I think that using a boolean to inform an error and return and error too if confusing.

Reply:
You can change this code to check for a specific error that `checkIfVulnarabilitiesHaveErrors` function returns. So for example, we can create an error like `var ErrVulnerabilitiesHasErrors = errors.new("vulnerabilities has errors")`, and return this error on function `checkIfVulnarabilitiesHaveErrors`, and them check if the error returned from analyzer is this error using `errors.Is(err, printresults.ErrVulnerabilitiesHasErrors)`. So we can change to something like this:
```
if err != nil && !errors.Is(err, printresults.ErrVulnerabilitiesHasErrors) { return err }

if errors.Is(err,  printresults.ErrVulnerabilitiesHasErrors) && s.configs.ReturnErrorIfFoundVulnerability {
...
}
```

Reply:
i may have make my variable inapropriate, i must chang it to hasUnkownVulnerability since error and this variable must be different on this case. Err is informing about some execution error and hasError is informing about and unknown vulnerabilities has been found. We are showing execution error even if the vulnerability is marked as RiskAccepted or FalsePositive. I will change the variable name to be more incisive. 
After the review i realize totalVulns isn't necessary, i will remove it and and make Analyze() return only boolean and error.

Reply:
[In Go we can treat  errors as values] (https://go.dev/blog/errors-are-values). So errors could be either unexpected execution error or business errors like this.